prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,552 @@
+                    processingModeFilter = listOf("true"),
+                    qualityFilter = filter, // it doesn't matter to the engine how the filter gets applied,
+                    reverseTheQualityFilter = reverseQuality
+                )
+            ),
+        )
+    }
+
+    private fun createFHIRFunctionsInstance(): FHIRFunctions {
+        val settings = FileSettings().loadOrganizations(universalPipelineOrganization)
+        val metadata = UnitTestUtils.simpleMetadata
+        metadata.lookupTableStore += mapOf(
+            "observation-mapping" to LookupTable("observation-mapping", emptyList())
+        )
+        val workflowEngine = WorkflowEngine
+            .Builder()
+            .metadata(metadata)
+            .settingsProvider(settings)
+            .databaseAccess(ReportStreamTestDatabaseContainer.testDatabaseAccess)
+            .build()
+        return FHIRFunctions(workflowEngine, databaseAccess = ReportStreamTestDatabaseContainer.testDatabaseAccess)
+    }
+
+    private fun createFHIRRouter(
+        org: DeepOrganization? = null,
+    ): FHIRRouter {
+        val settings = FileSettings().loadOrganizations(org ?: universalPipelineOrganization)
+        val metadata = UnitTestUtils.simpleMetadata
+        metadata.lookupTableStore += mapOf(
+            "observation-mapping" to LookupTable("observation-mapping", emptyList())
+        )
+        return FHIRRouter(
+            metadata,
+            settings,
+            reportService = ReportService(ReportGraph(ReportStreamTestDatabaseContainer.testDatabaseAccess)),
+        )
+    }
+
+    private fun generateQueueMessage(report: Report, blobContents: String, sender: Sender): String {
+        return """
+            {
+                "type": "route",
+                "reportId": "${report.id}",
+                "blobURL": "${report.bodyURL}",
+                "digest": "${BlobAccess.digestToString(BlobAccess.sha256Digest(blobContents.toByteArray()))}",
+                "blobSubFolderName": "${sender.fullName}",
+                "topic": "${sender.topic.jsonVal}",
+                "schemaName": "${sender.schemaName}" 
+            }
+        """.trimIndent()
+    }
+
+    private fun getBlobContainerMetadata(): BlobAccess.BlobContainerMetadata {
+        val blobConnectionString =
+            """DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=keydevstoreaccount1;BlobEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10000
+                )
+            }/devstoreaccount1;QueueEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10001
+                )
+            }/devstoreaccount1;"""
+        return BlobAccess.BlobContainerMetadata(
+            "container1",
+            blobConnectionString
+        )
+    }
+
+    private fun createReport(
+        fileFormat: Report.Format,
```

Comment:
does the route step ever receive an HL7 message? I thought everything got converted to FHIR by the preceding step?

Reply:
Why was this resolved? Comment seems unaddressed?

Reply:
that's on me - I thought the route step never received anything but FHIR hence I marked it as resolved. What would you like me to do here?

Reply:
You are correct, the route step doesn't receive anything but FHIR. I think what Michael is saying is we should resolve the IntelliJ warning somehow. One option is to not pass `fileFormat` in at all. I think I'm actually okay with the way it is now because I see createReport getting abstracted into a common location where other integration tests, like translation, can use it?

Reply:
reading this as no action on David's part

Reply:
✅